packages/react-reconciler/src/__tests__/ReactAsyncActions-test.js

```diff
@@ -1175,4 +1175,52 @@ describe('ReactAsyncActions', () => {
     await act(() => resolveText('Wait 2'));
     assertLog(['B']);
   });
+
+  // @gate enableAsyncActions
+  test('useOptimistic warns if outside of a transition', async () => {
+    let startTransition;
+    let setLoadingProgress;
+    let setText;
+    function App() {
+      const [, _startTransition] = useTransition();
+      const [text, _setText] = useState('A');
+      const [loadingProgress, _setLoadingProgress] = useOptimistic(0);
+      startTransition = _startTransition;
+      setText = _setText;
+      setLoadingProgress = _setLoadingProgress;
+
+      return (
+        <>
+          {loadingProgress !== 0 ? (
```

Comment:
Maybe make this unconditional so it’s easier to witness that the state gets set back? Right now the test logs sorta make it look like it gets stuck at 25.

Reply:
Wait never mind I'm silly. The toMatchRenderedOutput is plenty.